### PR TITLE
[MSKINS-188] improve maven logo

### PR DESCRIPTION
**Pull request by stevecrox**
Attempt to add an improved logo, checking this appears to work in the various views.

### CODE REVIEW

**Comment by michael-o** [comment]
Thanks, will check!

File: pom.xml [hunk 1]
```diff
@@ -160,6 +160,7 @@ under the License.
           <artifactId>apache-rat-plugin</artifactId>
           <configuration>
             <excludes combine.children="append">
+              <exclude>src/main/resources/images/logos/maven-feather.svg</exclude>
```
Comment (michael-o) [inline pom.xml:163]: You have added a PNG, but excluding a SVG?

Comment (stevecrox) [reply pom.xml:163]: Indeed, I've redone the commit

**Comment by stevecrox** [comment]
I left in maven-feather.png (and updated it to the modern version) because the Integration Tests reference it and the older buttons.

**Comment by slawekjaranowski** [comment]
@stevecrox if you start your commit message with issue in square brackets  should be linked with jira, 
like for this changes should be: [MSKINS-188]

File: pom.xml [hunk 2]
```diff
@@ -160,6 +160,8 @@ under the License.
           <artifactId>apache-rat-plugin</artifactId>
           <configuration>
             <excludes combine.children="append">
+              <exclude>src/main/resources/images/logos/maven-feather.png</exclude>
+              <exclude>src/main/resources/images/logos/powered-by-apache.png</exclude>
```
Comment (slawekjaranowski) [inline pom.xml:164]: Do you need exclude binary file?

**Comment by slawekjaranowski** [comment]
Do we need image `powered-by-apache.png` in original resource - `2450 × 2450 pixels`, file size is about `540 KB`?

**Comment by stevecrox** [comment]
The reason for replacement is the original is quite low resolution, picking a higher resolution image means it will scale down for smaller systems while looking good on 4K monitors as well.

I could create a couple different resolution images and then write CSS to set an element background image on a span element which I make window width dependent but the rationale there is to speed dial up style connections from loading the page.

The element itself is within a span which means a delay in loading won't affect page loading and should get alt text which would be displayed if there is an issue loading the image.

File: pom.xml (hunk 2 above)
Comment (stevecrox) [reply pom.xml:164]: I do not, let me redo the commits

**Comment by stevecrox** [comment]
Should be fixed, I was going through each commit to make sure they were limited to strictly only stuff that is required.

"row-fluid pft that doesn't relate to spans" he said to himself despite knowing spans are how the grid system work in Bootstrap 2 and then not checking after removing before his submitted the PR.

**Comment by slawekjaranowski** [comment]
> The reason for replacement is the original is quite low resolution, picking a higher resolution image means it will scale down for smaller systems while looking good on 4K monitors as well. I lack a 4k monitor so I've kept it simple and used the original image.
> 
> I could create a couple different resolution images and then write CSS to set an element background image on a span element which I make window width dependent but the rationale there is to speed dial up style connections from loading the page.
> 
> The element itself is within a span which means a delay in loading won't affect page loading and should get alt text which would be displayed if there is an issue loading the image. Hence the KISS

I have display with resolution 3072 × 1920 and powered by logo have effective resolution 116 x 116
So is scaled from 2450 × 2450 -> 116 x 116

I only think if we really need so big input image? I'm not expert on this subject.

**Comment by stevecrox** [comment]
So .. knowing its only 116 x 116 on a 4k screen gave me a target. 

One of the factors I've included is accessibility I have worked with several people with sight issues and a common seems to be a 48" monitor combined with a few different ways of adding zoom on top. While working on the bootstrap upgrades I've been checking layout by reducing it to <578px width (for phones) and also leaving a normal layout but zoomed to 400% (it results in a display size similar to my co-workers setup).

- Using Gwenview reducing it to 5% (122 x 122) cut the size down to 14Kib, then zooming into 400% it looked like a blocky mess.
- Using Gwenview reducing it to 10% (245 x 245) cut the size down to 26Kib, then zooming into 400% it starts getting blocky but looks ok.
- Using Gwenview reducing it to 15% (367 x 367) cut the size down to 44Kib, then zooming into 400% it looks pretty good but is larger than my screen

I did a quick build of the Integration Tests and in v2.8 the side bar takes the entire screen width so checking with Gwenview was pretty accurate.

Is there any chance you could get the image size at a 400% zoom? I'm inclined to go with the 10% sized image (which I've committed).

File: src/main/resources/META-INF/maven/site.vm [hunk 3]
```diff
@@ -222,14 +222,17 @@
...
-            <div id="poweredBy">
+            <div id="poweredBy" class="row-fluid" >
 #*        *##facebookLike( $sideBarEnabled )
               <div class="clear"></div>
 #*        *##followTwitter( $sideBarEnabled )
               <div class="clear"></div>
 #*        *##flattrBody( $sideBarEnabled )
               <div class="clear"></div>
-#*        *##builtByLogo( $decoration.poweredBy )
+              <!-- MSKINS-188, added to centralise the logo. -->
```
Comment (michael-o) [inline src/main/resources/META-INF/maven/site.vm:232]: Drop this line, we have commit messages

**Review by michael-o — CHANGES_REQUESTED** [review]

**Comment by slawekjaranowski** [comment]
Finally please squash to one commit.

**Comment by stevecrox** [comment]
Why?

Squashing should be used on long standing branches where people have made a mess of the commit chain (e.g. committing changes and a second commit to revert, or merging main into the branch periodically rather than rebasing).

But this is a feature branch from main, the resulting merge will ensure a continuous single chain of commits appear in the git changelog.

**Comment by michael-o** [comment]
> Why?
> 
> Squashing should be used on long standing branches where people have made a mess of the commit chain (e.g. committing changes and a second commit to revert, or merging main into the branch periodically rather than rebasing).
> 
> But this is a feature branch from main, the resulting merge will ensure a continuous single chain of commits appear in the git changelog.

No intermediate mess on master.

**Comment by stevecrox** [comment]
I don't see any reasoning in that statement and its phrased in an unnecessary way.

I've already explained sound reasons for squashing a commit. This project appears to be following some variation of [Feature Branch Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow). The key requirement of that approach is to ensure main (master is not inclusive) branch is always buildable and releasable.

The only reasons a commit in a branch matter are for the reasons I've outlined.

I'm not going to squash it and I figure you have the following choices.

- Provide a solid engineering based reason for squashing
- Accept the PR
- Reject the PR

If you accept it, I'll happily keep plugin away at updating the skin to Bootstrap v5 with all the UI aspects the project appears to value (I'm not sure I agree with a number of them some but whatever..).

If you reject it, I'll just fork and rename the project and release it as Bootstrap v5.

**Comment by slawekjaranowski** [comment]
Simply we require linear commit on master branch - from master perspective final result is important - we needn't history of work on PR in master.

Another reason is to have easy possibility to revert only one commit not many, if will needed.

I hope we don't waste your and our time for such discussion - please simply respect our easy requirement.

We can also accept PR and squash during merge.

**Comment by stevecrox** [comment]
I'm sorry picking one git hash value to roll back to over another is the same effort.

You've wasted me time, do it yourself I'm done with being insulted by the Maven team.

**Comment by stephenc** [comment]
For the record, as a Maven committer (who is not currently active) I disagree with the approach to merging. We could have merged this through a manual squash merge while retaining the author metadata.

To the author, the commit rework would mean that your commit is fully attributable to you as both committer and author which is something that would be nice to have but shouldn’t be a roadblock

**Comment by michael-o** [comment]
> For the record, as a Maven committer (who is not currently active) I disagree with the approach to merging. We could have merged this through a manual squash merge while retaining the author metadata.
> 
> To the author, the commit rework would mean that your commit is fully attributable to you as both committer and author which is something that would be nice to have but shouldn’t be a roadblock

Correct,  we always retain authorship.

**Comment by jira-importer** [comment]
Resolve #234

**Pull request closed without merging**